src/rust/bitbox02-rust/src/workflow/menu.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::bb02_async::option;
+use core::cell::RefCell;
+extern crate alloc;
+use alloc::boxed::Box;
+
+pub async fn menu() {
+    let result = RefCell::new(None);
+    let mut component = bitbox02::ui::menu_create(
+        &["foo", "bar"],
+        Some("demo title"),
+        Some(Box::new(|idx| {
+            *result.borrow_mut() = Some(idx);
+        })),
+        Some(Box::new(|| crate::print_debug!(1000, "hit continue"))),
+        Some(Box::new(|| crate::print_debug!(1000, "hit cancel"))),
```

Comment:
tbh it's not the easiest to read. would it make sense to define a `StatusComponent`, `MenuComponent`, etc. where I could do:

```rust
let mut menu = bitbox02::ui::Menu {
  words: &["foo", "bar"],
  title: Some("title"),
  on_select: Some(Box::New(|i| { *result.borrow_mut() = Some(i); })),
  on_cancel: ...,
  ...
}
menu.screen_stack_push(); // sets everything up and calls bitbox02_sys::menu_create(...)
```

so that it's clear what's what, especially which callback is which.

Reply:
I made a MenuParams struct similar to ConfirmParams, which gives each param a name. Should be more readable now, ptal.